backend/tournesol/suggestions/suggester.py

```diff
@@ -0,0 +1,236 @@
+from django.db.models import Q, QuerySet
+
+from core.models import User
+from tournesol.models import ComparisonCriteriaScore, Entity, Poll
+from tournesol.suggestions.graph import Graph
+from tournesol.suggestions.suggested_user import SuggestedUser as RecommendationUser
+from tournesol.suggestions.suggested_video import SuggestedVideo
+
+
+class Suggester:
+    """
+    An interface class to ask for suggestions of videos to compare
+    """
+    # Entity to specific video structure dictionary
+    _entity_to_video: dict[Entity, SuggestedVideo]  # todo Use only uid as key
+    _comparison_reference_video: SuggestedVideo
+    # Graph containing all videos and existing comparisons, used to get the video preferences
+    _complete_graph: Graph
+    # Dictionary linking a user to its comparison graph, used to get its information gain
+    _user_specific_graphs: dict[User, Graph]
+
+    def __init__(self, concerned_poll: Poll):
+        """
+        Function used to initialize the class
+        It must not be called before the DB is ready, as it will call it while constructing the
+        complete graph
+        """
+        self._comparison_reference_video = SuggestedVideo(None)
+        self._entity_to_video = {}
+        self._user_specific_graphs = {}
+        self.poll = concerned_poll
+        self.criteria = self.poll.criterias_list[0]
+        # build complete graph
+        comparison_queryset: QuerySet = ComparisonCriteriaScore.objects \
+            .filter(comparison__poll__name=self.poll.name) \
+            .filter(criteria=self.criteria)
+        self._complete_graph = Graph(None, self.poll, self.criteria)
+
+        for c in comparison_queryset:
+            # Checks if each compared Entity has already been translated to a SuggestedVideo object
+            # and translates it otherwise
+            if c.comparison.entity_1 not in self._entity_to_video.keys():
```

Comment:
[minor]
`.keys()` is not required here
```suggestion
            if c.comparison.entity_1 not in self._entity_to_video:
```

Reply:
It is not necessary, but isn't it better to be more explicit if it doesn't have cost on the execution ?